### PR TITLE
Add CHANGELOG for v1 API

### PR DESCRIPTION
**Pull request by banool**
## Description
This PR adds a CHANGELOG for the Aptos Node API.

## Test Plan
My CI script: https://gist.github.com/banool/fb3884baea29a911f000710c165831f5.

- - -
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/2539)

### CODE REVIEW

File: api/doc/v1/CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
+# Aptos Node API Changelog
+
+All notable changes to the Aptos Node API will be captured in this file. This changelog is written by hand for now. It adheres to the format set out by [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
+
+**Note**: The Aptos Node API does not follow semantic version while we are in active development. Instead, breaking changes will be announced with each devnet cut. Once we launch our mainnet, the API will follow semantic versioning closely.
+
+## 1.0.0 (2022-08-04)
```
Comment (rajkaramchedu) [inline api/doc/v1/CHANGELOG.md:7]: The only thing I ask in date formatting is to explicitly say `(04 August 2022)` which 100% avoids any ambiguity. Even after so many years in the good old USA, I still can't get used the idea of month first, for example.

**Review by rajkaramchedu — APPROVED** [review]

File: api/doc/v1/CHANGELOG.md (hunk 1 above)
Comment (banool) [reply api/doc/v1/CHANGELOG.md:7]: I'm a pretty firm believer in YYYY-MM-DD, I don't think anyone would confuse this for YYYY-DD-MM.

Comment (rajkaramchedu) [reply api/doc/v1/CHANGELOG.md:7]: Haha, okay then.

Comment (banool) [reply api/doc/v1/CHANGELOG.md:7]: Spoke offline, we'll keep YYYY-MM-DD.

**Pull request merged**